lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -2338,6 +2330,106 @@ createSpecializedStructOrClassType(NominalOrBoundGenericNominalType *Type,
     }
   }
 
+  /// Returns true if the type's mangled name is affected by an
+  /// @_originallyDefinedIn annotation. This annotation can be on the type
+  /// itself, one of its generic arguments, etc.
+  bool containsOriginallyDefinedIn(Type T) {
```

Comment:
@slavapestov I found the following situations where @_originallyDefinedIn could affect a type's mangled name:

- The type itself being annotated.
- The bound generic arguments being annotated.
-  A typealias local to a function could be affected by the types in the function's generic signature.
- A metatype's type.

Do you know of any other ones?